### PR TITLE
ESlint Prittier and Husky

### LINKED ISSUE #1: ESlint Prittier and Husky
- [x] ESlint

- [x] Prittier

- [x] Husky

### PR DESCRIPTION
**Pull request by kitravee**
I create Typescript Next project
and config ESlint, Prittier, and Husky

For more information : [Youtube Source](https://www.youtube.com/watch?v=sH93pQb9bWM&t=83s&ab_channel=JarrodWatts)

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo '🏗️👷 Styling, testing and building your project before committing'
+
+# Check Prettier standards
+npm run check-format ||
+(
+    echo '🤢🤮🤢🤮 Its FOKING RAW - Your styling looks disgusting. 🤢🤮🤢🤮
```
Comment (jbuican14) [inline .husky/pre-commit]: strong and direct communication. agreed ;)

File: .eslintrc [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+  "ignorePatterns": ["next-env.d.ts"],
+  "extends": [
+    "next",
+    "next/core-web-vitals",
+    "plugin:react/recommended",
+    "google",
```
Comment (jbuican14) [inline .eslintrc:18]: what is the google one for? api?

File: .eslintrc [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+  },
+  "plugins": ["react", "@typescript-eslint"],
+  "rules": {
+    "no-console": 1,
+    "require-jsdoc": "off",
+    "import/newline-after-import": ["error", { "count": 1 }]
+  },
+  "settings": {
+    "react": {
+      "version": "latest"
```
Comment (jbuican14) [inline .eslintrc]: will it break if we use latest instead of define a certain version? Just a thought.

**Review by jbuican14 — APPROVED** [review]

File: .eslintrc (hunk 2 above)
Comment (kitravee) [reply .eslintrc:18]: it is a google eslint coding style.

package.json at devDepenency
`"eslint-config-next": "11.0.1"`

*Note After we install the package, we have to push it into this array.

File: .eslintrc (hunk 3 above)
Comment (kitravee) [reply .eslintrc]: yes, I have to change it >  "version": "detect"
 // React version. "detect" automatically picks the version you have installed.
 
*Note [Ref Doc](https://www.npmjs.com/package/eslint-plugin-react)

**Review by jbuican14 — APPROVED** [review]
sounds good to me.

**Pull request merged**